src/query.c

```diff
@@ -446,7 +467,79 @@ extractClauseFromNullTest(Relids base_relids,
 	}
 }
 
+/*
+ *	Convert a "BoolExpr" (IS TRUE, IS FALSE, IS NOT TRUE, IS NOT FALSE)
+ *	to the corresponding qualifier.
+ */
+void extractClauseFromBooleanTest(Relids base_relids, BooleanTest *node, List **quals) 
+{
+	elog(DEBUG3, "entering extractClauseFromBooleanTest()");
+	if (IsA(node->arg, Var))
+	{
+		Var		   *var = (Var *) node->arg;
+		MulticornBaseQual *result;
+		char	   *opname = NULL;
+		Expr	   *val;
 
+		if (var->varattno < 1)
+		{
+			return;
+		}
+		switch (node->booltesttype)
+		{
+			case IS_TRUE:
+				opname = "IS";
+				val = (Expr *) makeConst(BOOLOID, -1, InvalidOid, sizeof(bool), BoolGetDatum(true), false, true);
+				break;
+			case IS_FALSE:
+				opname = "IS";
+				val = (Expr *) makeConst(BOOLOID, -1, InvalidOid, sizeof(bool), BoolGetDatum(false), false, true);
+				break;
+			case IS_NOT_TRUE:
+				opname = "IS NOT";
+				val = (Expr *) makeConst(BOOLOID, -1, InvalidOid, sizeof(bool), BoolGetDatum(true), false, true);
+				break;
+			case IS_NOT_FALSE:
+				opname = "IS NOT";
+				val = (Expr *) makeConst(BOOLOID, -1, InvalidOid, sizeof(bool), BoolGetDatum(true), false, true);
+				break;
+			default:
+				/* IS UNKNOWN, IS NOT UNKNOWN */
+				elog(ERROR, "unsupported boolean test type %d", node->booltesttype);
```

Comment:
I think that these cases shouldn't be errors -- they should just be skipped from being sent to the python layer as quals, so that they can still be implemented by Postgres.  Although since these cases seem to basically be the same as "IS NULL" and "IS NOT NULL", it would also be feasible to implement them.